example-app/App.tsx

```diff
@@ -40,6 +40,41 @@ const Header = () => {
 };
 
 const App = () => {
+  const [result, setResult] = React.useState<{
+    success: boolean | undefined;
+    errorReason: string | undefined;
+  }>({success: undefined, errorReason: undefined});
+
+  React.useEffect(() => {
+    Linking.addEventListener('url', event => {
+      setResult(handleDeepLink(event.url));
+    });
+
+    async function getDeepLink() {
+      let url = await Linking.getInitialURL();
+      let res = handleDeepLink(url);
+      setResult(res);
+    }
+
+    getDeepLink();
+  }, []);
+
+  const handleDeepLink = (
+    url: string | null,
+  ): {success: boolean | undefined; errorReason: string | undefined} => {
+    if (!url) {
+      return {success: undefined, errorReason: undefined};
```

Comment:
```suggestion
      return undefined;
```

Maybe better if you don't even create the object if there is no URL.